examples/cpp/benchmark/SubscriberApp.cpp

```diff
@@ -0,0 +1,399 @@
+    if (reader_ == nullptr)
+    {
+        throw std::runtime_error("DataReader initialization failed");
+    }
+}
+
+SubscriberApp::~SubscriberApp()
+{
+    if (nullptr != participant_)
+    {
+        // Delete DDS entities contained within the DomainParticipant
+        participant_->delete_contained_entities();
+
+        // Delete DomainParticipant
+        DomainParticipantFactory::get_instance()->delete_participant(participant_);
+    }
+}
+
+void SubscriberApp::on_publication_matched(
+        DataWriter* /*writer*/,
+        const PublicationMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        matched_ = static_cast<int16_t>(info.current_count);
+        std::cout << "Publisher matched." << std::endl;
+        cv_.notify_one();
+    }
+    else if (info.current_count_change == -1)
+    {
+        matched_ = static_cast<int16_t>(info.current_count);
+        std::cout << "Publisher unmatched." << std::endl;
+        stop();
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for PublicationMatchedStatus current count change" << std::endl;
+    }
+}
+
+void SubscriberApp::on_subscription_matched(
+        DataReader* /*reader*/,
+        const SubscriptionMatchedStatus& info)
+{
+    if (info.current_count_change == 1)
+    {
+        std::cout << "Subscriber matched." << std::endl;
+    }
+    else if (info.current_count_change == -1)
+    {
+        std::cout << "Subscriber unmatched." << std::endl;
+        stop();
+    }
+    else
+    {
+        std::cout << info.current_count_change
+                  << " is not a valid value for SubscriptionMatchedStatus current count change" << std::endl;
+    }
+}
+
+void SubscriberApp::on_data_available(
+        DataReader* reader)
+{
+    SampleInfo info;
+    switch (msg_size_)
+    {
+        case CLIParser::MsgSizeKind::NONE:
+            while ((!is_stopped()) && (RETCODE_OK == reader->take_next_sample(&benchmark_, &info)))
+            {
+                if ((info.instance_state == ALIVE_INSTANCE_STATE) && info.valid_data)
+                {
+                    std::cout << "Sample with index: '" <<
+                        benchmark_.index() << "' (Array 0 Bytes) RECEIVED" << std::endl;
+                    benchmark_.index(benchmark_.index() + 1);
+                    while (matched_ == 0)
+                    {
+                        std::unique_lock<std::mutex> initial_lock(mutex_);
+                        auto check = cv_.wait_for(initial_lock, std::chrono::milliseconds(1), [&]()
+                                        {
+                                            return is_stopped();
+                                        });
+                        if (check)
+                        {
+                            return;
+                        }
+                    }
+                    if (benchmark_.index() >= samples_)
```

Comment:
I think we are missing this here, otherwise when the example is launched without `--samples` the subscriber will receive the first sample and skip.
```suggestion
                    if (samples_ != 0 && benchmark_.index() >= samples_)
```